src/model.rs

```diff
@@ -29,31 +29,25 @@ pub fn to_yaml<T: Serialize>(instance: &T) -> Result<String> {
 /// Will return `Err` if there is an issue converting YAML files for spec+annotation into a model
 /// instance.
 pub fn from_yaml<T: DeserializeOwned>(
-    spec_yaml: &str,
     hash: &str,
+    spec_yaml: &str,
     annotation_yaml: Option<&str>,
 ) -> Result<T> {
-    let mut spec_mapping: BTreeMap<String, Value> = serde_yaml::from_str(spec_yaml)?;
-
-    // Insert annotation if there is something
-    if let Some(yaml) = annotation_yaml {
-        let annotation_map: Mapping = serde_yaml::from_str(yaml)?;
-        spec_mapping.insert("annotation".into(), Value::from(annotation_map));
+    let mut spec: BTreeMap<String, Value> = serde_yaml::from_str(spec_yaml)?;
+    spec.insert("hash".to_owned(), Value::from(hash));
+    if let Some(resolved_annotation_yaml) = annotation_yaml {
+        let annotation: Mapping = serde_yaml::from_str(resolved_annotation_yaml)?;
+        spec.insert("annotation".to_owned(), Value::from(annotation));
     }
-    spec_mapping.insert("hash".to_owned(), Value::from(hash));
 
-    Ok(serde_yaml::from_str(&serde_yaml::to_string(
-        &spec_mapping,
-    )?)?)
+    Ok(serde_yaml::from_str(&serde_yaml::to_string(&spec)?)?)
 }
 
 // --- core model structs ---
 
 /// A reusable, containerized computational unit.
-#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
```

Comment:
I would probably leave clone there because we will need it later chances are

Reply:
You are right but I'd rather add it with the feature that requires it.

For example, if implementing `PodJob` requires this, then it would make sense to me to add that in the PR that introduces `PodJob` so that the diff reflects the need for it and its application.